### PR TITLE
[POAE7-2659] logical op and bool type support

### PR DESCRIPTION
**Pull request by yma11**
### What changes were proposed in this pull request?
Add logical op &&/|| and bool type support

### Why are the changes needed?
expr codegen for nextgen

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UT

### CODE REVIEW

**Comment by yma11** [comment]
@jikunshang @bigPYJ1151 @kd97 please help take a review. Thanks.

File: cider/type/plan/BinaryExpr.cpp [hunk 1]
```diff
@@ -119,6 +118,54 @@ JITExprValue& BinOper::codegenFixedSizeColCmpFun(JITValuePointer& null,
...
+JITExprValue& BinOper::codegenFixedSizeLogicalFun(JITFunction& func,
+                                                  JITValuePointer& null,
+                                                  FixSizeJITExprValue& lhs_val,
+                                                  FixSizeJITExprValue& rhs_val) {
+  switch (get_optype()) {
+    case kAND: {
+      // If one side is not null and false, return not null and FALSE
+      // else no special handle needed
+      auto if_builder = func.createIfBuilder();
+      auto value = func.createVariable(JITTypeTag::BOOL, "logical_or_val");
```
[the remark this answers is not in the document]
Comment (yma11) [reply cider/type/plan/BinaryExpr.cpp:130]: yep. Will update it.

File: cider/type/plan/BinaryExpr.cpp [hunk 2]
```diff
@@ -119,6 +118,54 @@ JITExprValue& BinOper::codegenFixedSizeColCmpFun(JITValuePointer& null,
...
+      if_builder
+          ->condition([&]() {
+            auto condition = (!lhs_val.getNull() && !lhs_val.getValue()) ||
```
[the remark this answers is not in the document]
Comment (yma11) [reply cider/type/plan/BinaryExpr.cpp:133]: not quite. The logic is not so straight. Refer [doc](https://prestodb.io/docs/current/functions/logical.html)

File: cider/exec/nextgen/operators/RowToColumnNode.cpp [hunk 3]
```diff
@@ -97,6 +92,32 @@ class ColumnWriter {
...
   }
 
+  JITValuePointer setFixSizeRawData(utils::FixSizeJITExprValue& fixsize_val) {
+    if (expr_->get_type_info().get_type() == kBOOLEAN) {
+      auto raw_data_buffer = context_.getJITFunction()->createLocalJITValue(
+          [this]() { return allocateBitwiseBuffer(1); });
+      // leverage existing set_null_vector but need opposite value as input
+      // TODO: (yma11) need check in UT
+      context_.getJITFunction()->emitRuntimeFunctionCall(
+          "set_null_vector",
```
Comment (bigPYJ1151) [inline cider/exec/nextgen/operators/RowToColumnNode.cpp:102]: Could use ```cider_agg_id_proj_bool```.

File: cider/type/plan/BinaryExpr.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (bigPYJ1151) [reply cider/type/plan/BinaryExpr.cpp:133]: Seems logical function implemented as a branch model, I implemented it as a branchless model before. Maybe we should do some benchmark recently.

File: cider/exec/nextgen/operators/RowToColumnNode.cpp (hunk 3 above)
Comment (yma11) [reply cider/exec/nextgen/operators/RowToColumnNode.cpp:102]: `cider_agg_id_proj_bool` take `int8_t` as the value and a cast from 1bit is needed if call this method.

**Comment by yma11** [comment]
@bigPYJ1151 Branchless code is ported and commented for now. We can have a test and see which one better.

**Review by bigPYJ1151 — APPROVED** [review]

**Review by jikunshang — APPROVED** [review]

**Pull request merged**